release/pins/make.py

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python3
+import platform
+import re
+import subprocess
+from pathlib import Path
+
+here = Path(__file__).parent
+outfile = (here / f"requirements-{platform.system().lower()}.txt").absolute()
+
+if __name__ == "__main__":
+    subprocess.check_call([
+        "pip-compile",
+        "--generate-hashes", "--upgrade", "--allow-unsafe",
+        "-o", outfile,
+        "requirements.in",
+    ], cwd=here)
+
+    out = outfile.read_bytes()
+    out = re.sub(b"-e .+", b"# --- line removed by make.py ---", out)
+    out = re.sub(b"pip-compile --.+", b"./make.py  (on the same platform)", out)
+    outfile.write_bytes(out)
```

Comment:
Why do we install ourself as editable here? Shouldn't we build the wheel, then install it + deps, and then take the requirements.txt?

Reply:
Well we can only pin our dependencies, we can't pin mitmproxy itself. That hash changes with every commit. 😄 So in either case we need to doctor ourselves out of the generated file. 

The alternative would be to have all requirements in a separate requirements.txt file and load that file in setup.py, then we could ingest that requirements file here. But that means ugly file processing in setup.py for environment markers. I don't think this would be much better?